packages/@aws-cdk/aws-appmesh/lib/shared-interfaces.ts

```diff
@@ -88,22 +88,53 @@ export interface PortMapping {
 }
 
 /**
- * Represents the properties needed to define healthy and active listeners for nodes
+ * Represents timeouts for HTTP protocols.
  */
-export interface VirtualNodeListener {
+export interface HttpTimeout {
   /**
-   * Array of PortMappingProps for the listener
+   * Represents an idle timeout. The amount of time that a connection may be idle.
    *
-   * @default - HTTP port 8080
+   * @default - none
    */
-  readonly portMapping?: PortMapping;
+  readonly idle?: cdk.Duration;
 
   /**
-   * Health checking strategy upstream nodes should use when communicating with the listener
+   * Represents per request timeout.
    *
-   * @default - no healthcheck
+   * @default - 15 s
    */
-  readonly healthCheck?: HealthCheck;
+  readonly perRequest?: cdk.Duration;
+}
+
+/**
+ * Represents timeouts for GRPC protocols.
+ */
+export interface GrpcTimeout {
+  /**
+   * Represents an idle timeout. The amount of time that a connection may be idle.
+   *
+   * @default - none
+   */
+  readonly idle?: cdk.Duration;
+
+  /**
+   * Represents per request timeout.
+   *
+   * @default - 15 s
+   */
+  readonly perRequest?: cdk.Duration;
+}
+
+/**
+ * Represents timeouts for TCP protocols.
+ */
+export interface TcpTimeout {
+  /**
+   * Represents an idle timeout. The amount of time that a connection may be idle.
+   *
+   * @default - none
+   */
+  readonly idle?: cdk.Duration;
```

Comment:
I believe all these types are used only in the `virtual-node-listener.ts` file. Can you move them there please?